Utils/utils.R

```diff
@@ -40,31 +40,73 @@ create_db <- function(){
   
   # Apply each query.
   sapply(queries, function(x){
-    res <- dbSendStatement(
-      con,
-      paste(scan(x, sep = "\n", what = "character"), collapse = ""))
     
-    dbClearResult(res)
+    tryCatch({
+      rs <- dbSendStatement(
+        con,
+        paste(scan(x, sep = "\n", what = "character"), collapse = ""))
+    }, error = function(err) {
+      message <- paste("dbSendStatement",err)
+      message(message, .loggit = FALSE)
+      loggit("ERROR", message)
+      dbDisconnect(con)
+    })
+    
+    dbClearResult(rs)
   })
   
   dbDisconnect(con)
 }
 
 db_fun <- function(query){
+  errFlag <- FALSE
   con <- dbConnect(RSQLite::SQLite(), db_name)
-  dat <- dbGetQuery(con,query)  # this does SendQuery, Fetch and ClearResult all in one
+  tryCatch(
+    expr = {
+      rs <- dbSendQuery(con, query)
+    },
+    warning = function(warn) {
+      message <- paste0("db_fun warning:\n",query,"\nresulted in\n",warn)
+      message(message, .loggit = FALSE)
+      loggit("WARN", message)
+      errFlag <<- TRUE
+    },
+    error = function(err) {
+      message <- paste0("db_fun error:\n",query,"\nresulted in\n",err)
+      message(message, .loggit = FALSE)
+      loggit("ERROR", message)
+      dbDisconnect(con)
+      errFlag <<- TRUE
+    },
+    finally ={
+      if (errFlag) return(NULL) 
+    })
+  
+  dat <- dbFetch(rs)
+  dbClearResult(rs)
+  if (nrow(dat) == 0 & str_detect(strng, "'Select'") == FALSE) {
```

Comment:
Hi @Robert-Krajcik , I get the following error when running the app on this branch:

`Warning: Error in str_detect: object 'strng' not found`

Reply:
OK I'll take a look in a few minutes